### DISCUSSION TITLE
What's the best way to track usage of deprecated fields and types?

### DISCUSSION THREAD

**Discussion by ruudk** [webonyx/graphql-php#1092] [Q&A] [answered]
I want to start logging usage of deprecated fields and types. What's the best way to do this?

My idea was to wrap all field resolvers and inspect the 4rd `ResolverInfo` argument.

Then log when `$resolverInfo->fieldDefinition->deprecationReason` is not null.

But I feel there must be a better way to do this. Any suggestions?

**Comment by spawnia** [comment, marked as answer]
In Lighthouse this is implemented as a validation rule - see https://github.com/nuwave/lighthouse/blob/e09943d8abc786cd266f4837ade0239fb508164c/src/Deprecation/DetectDeprecatedUsage.php

**Reply by ruudk** [reply]
Clever, this works great! Thank you.
